### PR TITLE
[feat]: [CDS-77039]: Fixed Service Environment code pulse annotations

### PR DESCRIPTION
**Pull request by lovish1999**
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51750)

### CODE REVIEW

File: 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java [hunk 1]
```diff
@@ -52,8 +53,7 @@
 import lombok.AllArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 
-@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true,
-    components = {HarnessModuleComponent.CDS_APPROVALS, HarnessModuleComponent.CDS_SERVICE_ENVIRONMENT})
+@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_APPROVALS})
```
Comment (puthrayaharness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java:56]: This is Artifactss

File: 125-cd-nextgen/src/main/java/io/harness/cdng/servicenow/resources/service/ServiceNowResourceServiceImpl.java [hunk 2]
```diff
@@ -66,8 +67,8 @@
 import lombok.extern.slf4j.Slf4j;
 
 @CodePulse(module = ProductModule.CDS, unitCoverageRequired = true,
-    components = {HarnessModuleComponent.CDS_SERVERLESS, HarnessModuleComponent.CDS_SERVICE_ENVIRONMENT,
-        HarnessModuleComponent.CDS_ECS, HarnessModuleComponent.CDS_APPROVALS})
+    components = {HarnessModuleComponent.CDS_SERVERLESS, HarnessModuleComponent.CDS_ECS,
```
Comment (puthrayaharness) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/servicenow/resources/service/ServiceNowResourceServiceImpl.java:70]: This is just Approvals

File: 999-annotations/src/main/java/io/harness/annotations/dev/HarnessModuleComponent.java [hunk 3]
```diff
@@ -32,5 +32,9 @@ public enum HarnessModuleComponent {
   CDS_TEMPLATE_LIBRARY,
   CDS_TRADITIONAL,
   CDS_TRIGGERS,
-  CDS_SLACK
+  CDS_SLACK,
+  CDS_AWS_SAM,
```
Comment (puthrayaharness) [inline 999-annotations/src/main/java/io/harness/annotations/dev/HarnessModuleComponent.java:36]: Do we have the following components in CD Jira? If not please remove. 
Let the respective teams add them.

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

File: 999-annotations/src/main/java/io/harness/annotations/dev/HarnessModuleComponent.java (hunk 3 above)
Comment (lovish1999) [reply 999-annotations/src/main/java/io/harness/annotations/dev/HarnessModuleComponent.java]: okay. Removed them.

File: 125-cd-nextgen/src/main/java/io/harness/cdng/servicenow/resources/service/ServiceNowResourceServiceImpl.java (hunk 2 above)
Comment (lovish1999) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/servicenow/resources/service/ServiceNowResourceServiceImpl.java]: Okay.

File: 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java (hunk 1 above)
Comment (lovish1999) [reply 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java]: Okay

**Comment by lovish1999** [comment]
trigger messagecheck

**Review by puthrayaharness — APPROVED** [review]

**Review by vivekkarnatiharness — APPROVED** [review]

**Comment by lovish1999** [comment]
trigger review

**Review by abhinavmittal-harness — APPROVED** [review]
> Reviewed 46 of 57 files at r1, 11 of 11 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51750)* status: 57 file(s) to review. 3. unresolved discussions.

**Review by naman-goenka — APPROVED** [review]

**Review by abhinavmittal-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51750)* status: 57 file(s) to review. 1 change requests. 3. unresolved discussions. 

___
*[`999-annotations/src/main/java/io/harness/annotations/dev/HarnessModuleComponent.java` line 36 at r2](https://reviewable.io/reviews/harness/harness-core/51750#-Nca6QS8B-d2wA0Bk42A:-Nca6QS8B-d2wA0Bk42B:b-z54oq7) ([raw file](https://github.com/harness/harness-core/blob/2a9a535fbc9bf9a673c3d46981e836772152235e/999-annotations/src/main/java/io/harness/annotations/dev/HarnessModuleComponent.java#L36)):*
> ```Java
>   CDS_TRIGGERS,
>   CI_PLG_LICENSING,
>   CDS_SLACK
> ```

Please remove CDS_SLACK component as well

**Comment by lovish1999** [comment]
> > _[Reviewable](https://reviewable.io/reviews/harness/harness-core/51750)_ status: 57 file(s) to review. 1 change requests. 3. unresolved discussions.
> 
> _[`999-annotations/src/main/java/io/harness/annotations/dev/HarnessModuleComponent.java` line 36 at r2](https://reviewable.io/reviews/harness/harness-core/51750#-Nca6QS8B-d2wA0Bk42A:-Nca6QS8B-d2wA0Bk42B:b-z54oq7) ([raw file](https://github.com/harness/harness-core/blob/2a9a535fbc9bf9a673c3d46981e836772152235e/999-annotations/src/main/java/io/harness/annotations/dev/HarnessModuleComponent.java#L36)):_
> 
> > ```java
> >   CDS_TRIGGERS,
> >   CI_PLG_LICENSING,
> >   CDS_SLACK
> > ```
> 
> Please remove CDS_SLACK component as well

Sure. Since all uts have already run for this PR. I will make this change in another PR https://github.com/harness/harness-core/pull/51755

**Review by yogesh-chauhan — APPROVED** [review]

**Review by tchaurasiya — APPROVED** [review]

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 3 of 57 files at r1, 1 of 11 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51750)* status: 53 file(s) to review. 3. unresolved discussions.

**Review by archit-harness — APPROVED** [review]

**Pull request merged**